bokeh/layouts.py

```diff
@@ -315,9 +315,16 @@ def gridplot(
             toolbar_options = {}
         if 'toolbar_location' not in toolbar_options:
             toolbar_options['toolbar_location'] = toolbar_location
+
+        # Fixed sizing mode needs scale_width for the toolbar
+        # for layout to work correctly.
+        if sizing_mode == 'fixed':
+            toolbar_sizing_mode = 'scale_width'
```

Comment:
Maybe technically, but it functionally doesn't matter.

`above` and `below` are what's broken for fixed, `left` and `right` work just fine because they get assigned a hard constraint of their width = 30px. In the above/below case, you want the width to be the width of the plot, so setting it to scale width allows it to grow. 

Things are happy to grow down a web page, they're just not happy to grow across it, which is why you have to force the width, but not the height.

Hope that makes sense.